internal/downloader/s3.go

```diff
@@ -0,0 +1,70 @@
+package downloader
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+type downloaderAPI interface {
+	DownloadWithContext(ctx aws.Context, w io.WriterAt, input *s3.GetObjectInput, options ...func(*s3manager.Downloader)) (n int64, err error)
+}
+
+type Downloader struct {
+	Bucket string
+	d      downloaderAPI
+}
+
+func New(bucketName string, dl downloaderAPI) Downloader {
+
+	return Downloader{
+		Bucket: bucketName,
+		d:      dl,
+	}
+}
+
+func (dl Downloader) Download(ctx context.Context, key string) ([]byte, error) {
```

Comment:
As we discussed, it turns out that the `aws.WriteAtBuffer` type does not have a `Read()` method. Since this is the case, it makes more sense to leave the type signature as is.